### PR TITLE
Competing keywords execution being executed on links to self

### PR DESCRIPTION
**Pull request by jcomack**
Implemented a check to exclude external links to itself from being assessed for competing keywords

Fixes https://github.com/Yoast/wordpress-seo/issues/4348
Fixes https://github.com/Yoast/YoastSEO.js/issues/675

### CODE REVIEW

File: js/researches/getLinkStatistics.js [hunk 1]
```diff
@@ -21,6 +22,37 @@ var keywordInAnchor = function( keyword, anchor, locale ) {
...
+};
+
+/**
+ * Matches the URL retrieved from an anchor with the current url.
+ *
+ * @param {string} url The URL to look for.
+ * @param {string} currentUrl The current URL to match against.
+ * @returns {boolean} Whether or not the URLs match.
+ */
+var matchAnchorUrlToCurrentUrl = function( url, currentUrl ) {
```
Comment (atimmer) [inline js/researches/getLinkStatistics.js]: I would call this `isLinkToSelf`

File: js/stringProcessing/extractUrl.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Retrieves all URLs from a string, based on the found anchors.
```
Comment (atimmer) [inline js/stringProcessing/extractUrl.js]: Why would you try to match all href's in a URL, you are only throwing one anchor in this function so wouldn't it make more sense to only match once and just return '' or the matched url?

**Comment by atimmer** [comment]
CR Done 👷🏻

**Comment by jcomack** [comment]
Feedback has been processed. Regarding `extractUrl.js`, @atimmer and I came to an agreement to leave this as is for the time being.

**Comment by diedexx** [comment]
CR Done 👍

**Comment by terw-dan** [comment]
Acceptance:

My post has the URL: `http://local.wordpress.dev/banana` and keyword `banana`
If I link to this url with the keyword `banana` I still get the error I am linking to another page with my keyword.

**Comment by jcomack** [comment]
Thanks to @terw-dan we found out that the getLinkStatistics wasn't matching against a proper URL (only a slug).

Added functionality to the Wordpress SEO plugin that gets a permalink that needs to be used to fix this PR. Please review https://github.com/Yoast/wordpress-seo/pull/4998 prior to attempting this PR.

**Comment by rensw90** [comment]
CR-Done 👍

**Comment by rensw90** [comment]
Acceptance done, couldn't get it to work. Used the same input as Danny mentioned before.

**Comment by terw-dan** [comment]
CR Done :+1: 
Fixed the build by using the correct quotes.

**Comment by boblinthorst** [comment]
Acceptance: Still doesn't seem to work. Used same testcase as previous testers. (http://local.wordpress.dev/banana/ with keyword banana and linking to this url from an instance of the keyword in the content.

**Comment by boblinthorst** [comment]
acceptance ok 👍

**Pull request merged**